### PR TITLE
Option and key ("*") to collapse / expand all branches under PID 1

### LINKED ISSUE #68: Feature request: Collapse all children in tree view
I would like to get to the following view as soon as possible. Currently that means `f6` `down` `f6` `down` `..`

image

Solutions that I would be happy with:
- A shortcut to toggle all children of selected, for instance `shift` `f6`.
- Tree view have collapsed all by default (recursive). Expand will just expand the selected element, not recursive.

Relevant issues at original htop-repo:
- https://github.com/hishamhm/htop/issues/24 _(latest responses argue that it should be collapsed by default)_
- https://github.com/hishamhm/htop/pull/510 _(pull request for collapse all)_

### PR DESCRIPTION
**Pull request by fasterit**
... and PID 2 if kernel threads are shown

Based on hishamhm/htop#510 by Krishna Chaitanya, B
Closes #68

### CODE REVIEW

File: Action.c [hunk 1]
```diff
@@ -438,7 +447,7 @@ static const struct {
    { .key = "      H: ", .info = "hide/show user process threads" },
    { .key = "      K: ", .info = "hide/show kernel threads" },
    { .key = "      F: ", .info = "cursor follows process" },
-   { .key = "    + -: ", .info = "expand/collapse tree" },
+   { .key = "  + - *: ", .info = "expand/collapse tree" },
```
Comment (BenBE) [inline Action.c:450]: Include some indication, that <kbd>*</kbd> does all branches, in contrast to the other two keys.

File: DisplayOptionsPanel.c [hunk 2]
```diff
@@ -98,6 +98,7 @@ DisplayOptionsPanel* DisplayOptionsPanel_new(Settings* settings, ScreenManager*
    Panel_setHeader(super, "Display options");
    Panel_add(super, (Object*) CheckItem_newByRef("Tree view", &(settings->treeView)));
    Panel_add(super, (Object*) CheckItem_newByRef("- Tree view is always sorted by PID (htop 2 behavior)", &(settings->treeViewAlwaysByPID)));
+   Panel_add(super, (Object*) CheckItem_newByRef("- Tree view is collapsed by default", &(settings->allBranchesCollapsed)));
```
Comment (BenBE) [inline DisplayOptionsPanel.c:101]: Options are cheap …

File: TESTPLAN [hunk 3]
```diff
@@ -55,7 +55,9 @@ Main screen:
 
       <l> - enter LSOF screen.
 
-      <m>, <n>, <o>, <p> - do nothing.
+      <m>, <n>, <o> - do nothing.
+
+      <p> - hide/show program path.
```
Comment (BenBE) [inline TESTPLAN:60]: Usually subject to another PR ;-)

**Review by BenBE — COMMENTED** [review]
LGTM.

File: TESTPLAN (hunk 3 above)
Comment (fasterit) [reply TESTPLAN:60]: Came in from the hishamhm/htop#510 one, I shall accept no blame for poking a dead test script :-)

Comment (BenBE) [reply TESTPLAN:60]: Just put it in a separate commit and I'm totally fine with this change.

My complaint is basically about mixing unrelated changes in the same commit.

Comment (fasterit) [reply TESTPLAN:60]: Spliced it out to c44b2ec

**Comment by fasterit** [comment]
Should work on FreeBSD now as well

File: ProcessList.c [hunk 4]
```diff
@@ -494,6 +494,15 @@ void ProcessList_expandTree(ProcessList* this) {
    }
 }
 
+void ProcessList_collapseAllBranches(ProcessList* this) {
+   int size = Vector_size(this->processes);
+   for (int i = 0; i < size; i++) {
+      Process* process = (Process*) Vector_get(this->processes, i);
+      // FreeBSD has pid 0 = kernel and pid 1 = init, so init has tree_depth = 1
+      if (process->tree_depth > 0 && process->pid > 1) process->showChildren = false;
```
Comment (BenBE) [inline ProcessList.c:502]: Please line-wrap the body of that if statement.

File: htop.c [hunk 5]
```diff
@@ -469,6 +469,8 @@ int main(int argc, char** argv) {
    millisleep(75);
    ProcessList_scan(pl, false);
 
+   if (settings->allBranchesCollapsed) ProcessList_collapseAllBranches(pl);
```
Comment (BenBE) [inline htop.c:472]: Please wrap body to new line after if statement.

**Review by BenBE — CHANGES_REQUESTED** [review]

File: ProcessList.c (hunk 4 above)
Comment (fasterit) [reply ProcessList.c]: done

**Review by BenBE — APPROVED** [review]

**Pull request merged**